realm/src/androidTest/java/io/realm/DynamicRealmListTest.java

```diff
@@ -0,0 +1,204 @@
+
+import io.realm.dynamic.DynamicRealmList;
+import io.realm.dynamic.DynamicRealmObject;
+import io.realm.entities.AllJavaTypes;
+import io.realm.entities.Dog;
+
+public class DynamicRealmListTest extends AndroidTestCase {
+
+    private Realm realm;
+    private DynamicRealmObject dynamicObject;
+    private DynamicRealmList dynamicList;
+
+    @Override
+    protected void setUp() throws Exception {
+        RealmConfiguration realmConfig = new RealmConfiguration.Builder(getContext()).schema(AllJavaTypes.class, Dog.class).build();
+        Realm.deleteRealm(realmConfig);
+        realm = Realm.getInstance(realmConfig);
+        realm.beginTransaction();
+        AllJavaTypes obj = realm.createObject(AllJavaTypes.class);
+        obj.setFieldString("str");
+        obj.setFieldShort((short) 1);
+        obj.setFieldInt(1);
+        obj.setFieldLong(1);
+        obj.setFieldFloat(1.23f);
+        obj.setFieldDouble(1.234d);
+        obj.setFieldBinary(new byte[]{1, 2, 3});
+        obj.setFieldBoolean(true);
+        obj.setFieldDate(new Date(1000));
+        obj.setFieldObject(obj);
+        obj.getFieldList().add(obj);
+        dynamicObject = new DynamicRealmObject(obj);
+        dynamicList = dynamicObject.getList(AllJavaTypes.FIELD_LIST);
+        realm.commitTransaction();
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        super.tearDown();
+        realm.close();
+    }
+
+    public void testAddNullObjectThrows() {
+        realm.beginTransaction();
+        try {
+            dynamicList.add(null);
+            fail();
+        } catch (IllegalArgumentException ignored) {
+        } finally {
+            realm.cancelTransaction();
+        }
+    }
+
+    public void testAddWrongTableClassThrows() {
+        realm.beginTransaction();
+        Dog dog = realm.createObject(Dog.class);
+        try {
+            dynamicList.add(new DynamicRealmObject(dog));
+            fail();
+        } catch (IllegalArgumentException ignored) {
+        } finally {
+            realm.cancelTransaction();
+        }
+    }
+
+    public void testAddFromWrongRealmThrows() {
+        RealmConfiguration otherConfig = new RealmConfiguration.Builder(getContext()).name("realm2").build();
+        Realm.deleteRealm(otherConfig);
+        Realm realm2 = Realm.getInstance(otherConfig);
+        realm2.beginTransaction();
+        AllJavaTypes realm2Object = realm2.createObject(AllJavaTypes.class);
+        realm2.commitTransaction();
+
+        realm.beginTransaction();
+        try {
+            dynamicList.add(new DynamicRealmObject(realm2Object));
+            fail();
+        } catch (IllegalArgumentException ignored) {
+        } finally {
+            realm.cancelTransaction();
+            realm2.close();
+        }
+    }
+
+    public void testAddObject() {
+        realm.beginTransaction();
+        AllJavaTypes obj = realm.createObject(AllJavaTypes.class);
+        dynamicList.add(new DynamicRealmObject(obj));
```

Comment:
Checking that `dynamicList` only has one element before`add` would ensure that `add` only adds one element.